### PR TITLE
Remove mars Eclipse 4.5 as a target

### LINKED ISSUE #2557: Drop support for Mars
This issue is, for now, for tracking thoughts. If this is going to cause major problems for anyone, please do comment below.

When Mars usage drops low enough, eliminate support for it.

Disadvantages:

  * Lose anyone still using Java 7
  * Lose anyone still committed to Mars

Advantages:

  * Can move appengine-plugins-core and rest of build to Java 8
  * Moving to Java 8 lets us move to Tycho 1.0
  * cut 1/3 of our Travis CI and flakes
  * Can use Submonitor.split() instead of newChild
  * LDP Dockertools requires Neon: https://marketplace.eclipse.org/content/eclipse-docker-tooling#group-details
  * Bring Guava to 21 or later
  * `private final WritableList/* <GcpProject> */ input; // Generics supported only in Neon+`

[LSP support](https://developers.redhat.com/blog/2016/11/24/deliver-support-for-new-languages-in-eclipse-ide-faster-with-generic-editor-and-language-servers/) is also interesting but would require us to drop Neon too.

Maybe when Photon is released in 2018?

### PR DESCRIPTION
**Pull request by elharo**
fix #2557

### CODE REVIEW

**Comment by elharo** [comment]
```
955 [INFO] Computing target platform for MavenProject: com.google.cloud.tools.eclipse:google-cloud-plugin-eclipse.repo:0.1.0-SNAPSHOT @ /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/gcp-repo/pom.xml
6265 [ERROR] Internal error: org.eclipse.tycho.artifacts.TargetPlatformFilterSyntaxException: Failed to parse version range: [${jettyMinVersion},${jettyMaxVersion}): Format "format(n[.n=0;[.n=0;[.S=[A-Za-z0-9_-];='';]]])" was unable to parse ${jettyMinVersion} -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: org.eclipse.tycho.artifacts.TargetPlatformFilterSyntaxException: Failed to parse version range: [${jettyMinVersion},${jettyMaxVersion})
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:122)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:55)
Caused by: org.eclipse.tycho.artifacts.TargetPlatformFilterSyntaxException: Failed to parse version range: [${jettyMinVersion},${jettyMaxVersion})
```

File: pom.xml [hunk 1]
```diff
@@ -495,10 +418,12 @@
...
     <profile>
       <id>build-eclipse-oxygen</id>
-      <activation>
-        <property>
-          <name>eclipse.target</name> <value>oxygen</value>
-        </property>
+      <activation>		
+        <!-- Default target if no eclipse.target specified. -->		
+        <property>		
+          <name>!eclipse.target</name>
```
Comment (briandealwis) [inline pom.xml]: There are some odd tabs and whitespace here

File: .travis.yml [hunk 2]
```diff
@@ -16,9 +16,8 @@ install:
 jdk: oraclejdk8
 env:
   matrix:
-    # ECLIPSE_TARGET is the targeted platform: mars is the default target
+    # ECLIPSE_TARGET is the targeted platform: oxygen is the default target
     # JAVA_RUNTIME specifies the execution runtime for tests; must match an <id> in .travisci/toolchains.xml
-    - JAVA_RUNTIME=JavaSE-1.7
     - JAVA_RUNTIME=JavaSE-1.8 ECLIPSE_TARGET=neon
     - JAVA_RUNTIME=JavaSE-1.8 ECLIPSE_TARGET=oxygen MAVEN_FLAGS=-Pjacoco
```
Comment (briandealwis) [inline .travis.yml]: Need to remove `ECLIPSE_TARGET=oxygen`, since that's the default.

Comment (briandealwis) [reply .travis.yml]: And need to remove `-Declipse.target=oxygen` from `kokoro/windows/continuous.bat`.

**Pull request merged**